server/athenian/api/cache.py

```diff
@@ -215,28 +215,38 @@ def cache_key(*args, **kwargs) -> bytes:
 def cached_methods(cls):
     """Decorate class to properly support cached instance methods."""
     for name, func in inspect.getmembers(cls, predicate=inspect.isfunction):
-        if getattr(func, "__cached__", False) and not getattr(func, "__cached_method__", False):
-            def generate_cls_method_shim(func):
-                def cls_method_shim(self):
-                    async def cached_wrapper(*args, **kwargs):
-                        return await func(self, *args, **kwargs)
+        outer_func = func
 
-                    async def reset_cache(*args, **kwargs):
-                        return await func.reset_cache(self, *args, **kwargs)
+        while not getattr(func, "__cached__", False) or getattr(func, "__cached_method__", False):
+            # This is needed in case the `@cached` decorator is not the outermost one
+            func = getattr(func, "__wrapped__", None)
+            if not func:
+                break
```

Comment:
`:=` is awesome
```suggestion
            if not (func := getattr(func, "__wrapped__", None)):
                break
```

Reply:
Still need to get used to this idiom, I guess this means also that it's been a long time since the last time I seriously programmed in go 😂